nfs/init_nfs_server.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright © 2025 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nfs
+
+import (
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// init_nfs_server uses systemctl to initialize an nfs server if necessary
+func (cs *CsiNfsService) initializeNfsServer() error {
+	log.Infof("checking status of nfs-server")
+	// Check to see if nfs-server is active (exited ok)
+	// Note: systemctl doesn't work when invoked from a container unless you ssh localhost before executing it.
+	// You will get the message "Failed to connect to bus: No data available" because systemctl must think it's on the local host.
+	restartNfsServer := false
+	out, err := cs.executor.ExecuteCommand("chroot", "/noderoot", "ssh", "localhost", "systemctl", "status", "nfs-server")
+	if err != nil || !strings.Contains(string(out), "Active: active") {
+		log.Infof("nfs-server not active: %s", string(out))
+		restartNfsServer = true
+	}
+	out, err = cs.executor.ExecuteCommand("chroot", "/noderoot", "ssh", "localhost", "systemctl", "status", "nfs-mountd")
+	if err != nil || !strings.Contains(string(out), "Active: active") {
+		log.Infof("nfs-mountd not active: %s", string(out))
+	}
+
+	// Reinitialize the NFS server if necessary
+	if !restartNfsServer {
+		return nil
+	}
+
+	// First copy the nfs.conf file to /noderoot/etc/nfs.conf. This is a 2-step process.
+	log.Infof("Configuring /etc/nfs.conf")
+	out, err = cs.executor.ExecuteCommand("cp", "/nfs.conf", "/noderoot/tmp/nfs.conf")
```

Comment:
pardon my ignorance here, but where is the /nfs.conf coming from ? On typical systems, there is only /etc/nfs.conf, but that is not being modified only later.

Reply:
Unsure if this cp is needed. Based on the logs : 
{"level":"info","msg":"Configuring /etc/nfs.conf","time":"2025-03-19T10:09:44.404469155Z"}
{"level":"error","msg":"Couldn't copy /nfs.conf to /noderoot/tmp/nfs.conf: cp: cannot stat '/nfs.conf': No such file or directory\n","time":"2025-03-19T10:09:44.407121566Z"}

Reply:
So looking at little bit more into this, it looks like we are missing the file `nfs.conf` within the pod itself.

That being said, this failure doesn't stop the functionality but it was more for ease of configuration if the worker node wasn't setup with an `nfs_server`.

I think that resolving this should be handled in a different PR since
1. It doesn't affect functionality
2. Further testing would be needed to get this configuring correctly.
3. Is this needed in the long term or it should be part of documentation.

Thoughts?

Reply:
Yeah, it was more of a question based on review. We can discuss about this further offline. you can mark this as resolved.